BraveWallet/Crypto/Stores/SendTokenStore.swift

```diff
@@ -138,21 +150,34 @@ public class SendTokenStore: ObservableObject {
     let weiFormatter = WeiFormatter(decimalFormatStyle: .decimals(precision: 18))
     guard let token = selectedSendToken, let weiHexString = weiFormatter.weiString(from: amount, radix: .hex, decimals: 18) else { return }
     
-    if token.isETH {
-      let data = BraveWallet.TxData(nonce: "", gasPrice: "", gasLimit: "", to: address, value: "0x\(weiHexString)", data: .init())
-      transactionController.addUnapprovedTransaction(data, from: account.address) { success, txMetaId, errorMessage in
-        completion(success)
-      }
-    } else {
-      transactionController.makeErc20TransferData(account.address, amount: "0x\(weiHexString)") { [self] success, data in
-        guard success else {
-          completion(false)
-          return
-        }
-        rpcController.chainId { [self] chainId in
-          let txData = BraveWallet.TxData(nonce: "", gasPrice: "", gasLimit: "", to: token.contractAddress, value: "0x0", data: data)
-          transactionController.addUnapprovedTransaction(txData, from: account.address) { success, txMetaId, errorMessage in
-            completion(success)
+    rpcController.chainId { [weak self] chainId in
+      guard let self = self else { return }
+      self.rpcController.allNetworks { networks in
+        guard let currentNetwork = networks.first(where: { $0.id == chainId }) else { return }
+        
+        if token.isETH {
+          let baseData = BraveWallet.TxData(nonce: "", gasPrice: "", gasLimit: "", to: address, value: "0x\(weiHexString)", data: .init())
+          if currentNetwork.isEip1559 {
+            self.makeEIP1559Tx(chainId: chainId, baseData: baseData, from: account, completion: completion)
```

Comment:
Do we need to fall back to a regular transaction creation if this fails?

Reply:
i don't think so. add unapproved tx call should just base on the chain not the response of the api call.